src/workerd/server/tests/python/python-rpc/worker.py

```diff
@@ -0,0 +1,229 @@
+        )
+        assert_equal(received, val.to_py(), True)
+        received = await env.JsRpc.identity(val)
+        assert not isinstance(received, JsProxy), (
+            "Expected the returned value from RPC to be a Python type."
+        )
+        assert_equal(received, val.to_py(), True)
+
+    for val in [
+        js.ArrayBuffer.new(8),
+        js.DataView.new(js.ArrayBuffer.new(16)),
+        js.Int16Array.of("10", "24"),
+    ]:
+        js_binary = await env.PythonRpc.identity(val)
+        assert isinstance(js_binary, memoryview)
+        js_binary = await env.JsRpc.identity(val)
+        assert isinstance(js_binary, memoryview)
+
+    js_map = await env.PythonRpc.identity(
+        js.Map.new(
+            [
+                [1, "one"],
+                [2, "two"],
+                [3, "three"],
+            ]
+        ),
+    )
+    assert isinstance(js_map, dict)
+    assert js_map[1] == "one"
+
+    js_set = await env.PythonRpc.identity(
+        js.Set.new([1, 2, 3, 4]),
+    )
+    assert isinstance(js_set, set)
+    assert 1 in js_set
+    assert 42 not in js_set
+
+    js_undefined = await env.PythonRpc.identity(js.undefined)
+    assert js_undefined is None
+
+    js_date = await env.PythonRpc.identity(js.Date.new())
+    assert isinstance(js_date, datetime)
+
+    js_date_list = await env.PythonRpc.identity([js.Date.new(), js.Date.new()])
+    for d in js_date_list:
+        assert isinstance(d, datetime)
+
+    js_exception = await env.PythonRpc.identity(js.Error.new("message"))
+    assert isinstance(js_exception, Exception)
+
+    js_obj = await env.PythonRpc.identity(
+        to_js({"foo": 42}, dict_converter=js.Object.fromEntries)
+    )
+    assert isinstance(js_obj, dict)
+    assert js_obj["foo"] == 42
+
+    # Web/API Types
+    # - Response
+    py_response = await env.PythonRpc.handle_response(Response("this is a response"))
+    assert isinstance(py_response, Response)
+    assert await py_response.text() == "this is a response"
+    js_response = await env.JsRpc.handleResponse(Response("this is a response"))
+    assert await js_response.text() == "this is a response"
+    assert isinstance(js_response, Response)
+
+    # - Request
+    py_response = await env.PythonRpc.handle_request(
+        Request("https://test.com", method=HTTPMethod.POST)
+    )
+    assert isinstance(py_response, Request)
+    assert py_response.method == "POST"
+    js_response = await env.JsRpc.handleRequest(
+        Request("https://test.com", method=HTTPMethod.POST)
+    )
+    assert js_response.method == "POST"
+    assert isinstance(js_response, Request)
+
+    # - Verify that a JS type can be sent.
+    py_response2 = await env.PythonRpc.handle_response(js.Response.new("a JS response"))
+    assert await py_response2.text() == "a JS response"
+
+    # Verify that sending unsupported types fails.
+    await assertRaises(lambda: env.PythonRpc.one_arg(CustomType(42)))
```

Comment:
If you use `assertRaises` from `unittest` you can do:
```suggestion
    with assertRaises(ExpectedErrorType):
       env.PythonRpc.one_arg(CustomType(42))
```